### PR TITLE
added json coverage for scheduler

### PR DESCRIPTION
**Pull request by rahulii**
This PR expands JSON e2e tests to cover all control plane components(scheduler in this PR )

Reference : https://github.com/kubernetes/kubernetes/issues/100748

### CODE REVIEW

File: hack/ci/e2e-k8s.sh
Comment (serathius) [inline hack/ci/e2e-k8s.sh:89]: You need to pass this variable to scheduler config :P 
line 150
```
  scheduler:
    extraArgs:
        "v": "${KIND_CLUSTER_LOG_LEVEL}"
 ```

Comment (rahulii) [reply hack/ci/e2e-k8s.sh:89]: Done , please check!

File: hack/ci/e2e-k8s.sh
Comment (serathius) [inline hack/ci/e2e-k8s.sh:99]:
```suggestion
        \"logging-format\": \"${SCHEDULER_LOG_FORMAT}\""
```

File: hack/ci/e2e-k8s.sh
Comment (BenTheElder) [inline hack/ci/e2e-k8s.sh:99]: having trouble checking the indentation on github, the indentation here needs to match how indented it needs to be in the config.

File: hack/ci/e2e-k8s.sh
Comment (BenTheElder) [inline hack/ci/e2e-k8s.sh:90]: I think we should probably just phase out the KUBELET_LOG_FORMAT in favor of like CLUSTER_LOG_FORMAT and apply it to as many components as possible for the current version?

at some point we want to eliminate this script in favor of a real tool, expanding the "API" surface will make it less managable.

Comment (rahulii) [reply hack/ci/e2e-k8s.sh:90]: do you mean replacing KUBELET_LOG_FORMAT and SCHEDULER_LOG_FORMAT with CLUSTER_LOG_FORMAT?

Comment (rahulii) [reply hack/ci/e2e-k8s.sh:90]: /cc @BenTheElder

Comment (BenTheElder) [reply hack/ci/e2e-k8s.sh:90]: yes, but possible also support KUBELET_LOG_FORMAT until we can migrate CI.

sorry we were on holiday here, also I'm going to have limited availability the next ~two weeks

**Comment by BenTheElder** [comment]
/lgtm cancel
/approve
(this will allow someone else to LGTM once changes are made and have this merge)

**Review by serathius — APPROVED** [review]

**Pull request merged**
